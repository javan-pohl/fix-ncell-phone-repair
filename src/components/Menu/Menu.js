import React from 'react'

import { Link, useStaticQuery, graphql } from 'gatsby'
import { menu, open } from './Menu.module.css'

function getLinks() {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
          menuLinks {
            link
            testid
            text
          }
        }
      }
    }
  `)
  const {
    site: {
      siteMetadata: { menuLinks },
    },
  } = data
  return menuLinks
}
function Menu({ visible, close }) {
  const menuLinks = getLinks()
  const links = menuLinks.map((link) => (
    <Link data-testid={link.testid} to={link.link} key={link.testid}>
      <h1>{link.text}</h1>
    </Link>
  ))
  return (
    <div
      id='navMenu'
      data-testid='navMenu'
      role='button'
      tabIndex='0'
      /* we don't want to destructure props here because that will cause name overlap */
      /* eslint-disable */
      className={`${menu} ${visible ? `${open}` : ''}`}
      /* eslint-enable */
      onClick={() => close()}
      onKeyDown={() => close()}
    >
      {links}
    </div>
  )
}

export default Menu
