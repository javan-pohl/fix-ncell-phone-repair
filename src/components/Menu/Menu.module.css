.menu {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #ffffff;
  z-index: 1031;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  transition: all 0.35s;

  cursor: pointer;
}

a {
  pointer-events: all;
}