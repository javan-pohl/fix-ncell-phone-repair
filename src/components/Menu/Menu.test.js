import 'jsdom-global/register'
import React from 'react'
import * as Gatsby from 'gatsby'
import Menu from './Menu'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'


const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      title: 'FixNcell Phone Repair',
      menuLinks: [
        {
          testid: 'homeLink',
          link: '/',
          text: 'Home',
        },
        {
          testid: 'aboutLink',
          link: '/about',
          text: 'About',
        },
        {
          testid: 'repairsLink',
          link: '/iphone-repair',
          text: 'Repairs',
        },
        {
          testid: 'contactLink',
          link: '/contact',
          text: 'Contact Us',
        },
      ],
    },
  },
}))

beforeEach(() => {
  jest.clearAllMocks()
})

test('should not be visible when fed a "visible" prop of false', () => {
  render(<Menu visible={false} />)
  const menu = screen.getByTestId('navMenu')
  const menuEl = document.getElementById('navMenu')
  expect(menuEl).not.toBeVisible()
})

test('should be visible when fed a "visible" prop of true', () => {
  render(<Menu visible={true} />)
  const menu = screen.getByTestId('navMenu')
  const menuEl = document.getElementById('navMenu')
  expect(menuEl).toBeVisible()
})

describe('Menu', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Menu visible={true}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})