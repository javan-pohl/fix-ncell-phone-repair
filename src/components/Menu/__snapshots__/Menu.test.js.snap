// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<div
  className="menu"
  data-testid="navMenu"
  id="navMenu"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  style={
    Object {
      "opacity": "1",
      "visibility": "visible",
    }
  }
  tabIndex="0"
>
  <a
    data-testid="homeLink"
    href="/"
  >
    <h1>
      Home
    </h1>
  </a>
  <a
    data-testid="aboutLink"
    href="/about"
  >
    <h1>
      About
    </h1>
  </a>
  <a
    data-testid="repairsLink"
    href="/iphone-repair"
  >
    <h1>
      Repairs
    </h1>
  </a>
  <a
    data-testid="contactLink"
    href="/contact"
  >
    <h1>
      Contact Us
    </h1>
  </a>
</div>
`;
